### PR TITLE
Add Fedora 23 runs

### PR DESCRIPTION
**Pull request by ellismg**

### CODE REVIEW

**Comment by ellismg** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -26,6 +27,7 @@ def osGroupMap = ['Ubuntu14.04':'Linux',
 def targetNugetRuntimeMap = ['OSX' : 'osx.10.10-x64',
                              'Ubuntu14.04' : 'ubuntu.14.04-x64',
                              'Ubuntu16.04' : 'ubuntu.14.04-x64',
+                             'Fedora23' : 'ubuntu.14.04-x64',
```
Comment (bartonjs) [inline netci.groovy]: No Fedora-specific identifier here?

**Comment by bartonjs** [comment]
Test Innerloop Windows_NT Release Build and Test please (#7630)

**Comment by ellismg** [comment]
I will need to make a similar change in CoreCLR before this can work end to end (since we'll pick up CoreCLR from there).

**Comment by ellismg** [comment]
@Priya91 can you take a look?

File: netci.groovy (hunk 1 above)
Comment (ellismg) [inline netci.groovy]: We don't publish fedora packages (yet) so restore would fail.  I'll address this later this week.

**Comment by stephentoub** [comment]
LGTM

**Comment by Priya91** [comment]
LGTM

**Pull request merged**